server/src/main/java/org/opensearch/repositories/RepositoriesService.java

```diff
@@ -149,6 +149,19 @@ public RepositoriesService(
         deleteRepositoryTaskKey = clusterService.registerClusterManagerTask(ClusterManagerTaskKeys.DELETE_REPOSITORY_KEY, true);
     }
 
+    public RepositoriesService repositories(Map<String, Repository> repositories) {
+        this.repositories = repositories;
+        return this;
+    }
+
+    public Map<String, Repository> repositories() {
+        return this.repositories;
+    }
+
+    public Map<String, Repository.Factory> typesRegistry() {
```

Comment:
why did you expose these methods? We dont allow updating repos like this.

Reply:
I had to mock repositoriesService object, which was needed for the tests; else all the tests were reporting to nukl check failure for empty repositories: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/repositories/RepositoriesService.java#L489.

Reply:
```
 public Repository repository(String repositoryName) {
        Repository repository = repositories.get(repositoryName);
        if (repository != null) {
            return repository;
        }
        repository = internalRepositories.get(repositoryName);
        if (repository != null) {
            return repository;
        }
        throw new RepositoryMissingException(repositoryName);
    }
```

Reply:
Ack.